mysql/cloudmysql/cloudmysql.go

```diff
@@ -13,92 +13,130 @@
 // limitations under the License.
 
 // Package cloudmysql provides connections to managed MySQL Cloud SQL instances.
+// See https://cloud.google.com/sql/docs/mysql/ for more information.
+//
+// URLs
+//
+// For mysql.Open, cloudmysql registers for the scheme "cloudmysql".
+// The default URL opener will create a connection using the default
+// credentials from the environment, as described in
+// https://cloud.google.com/docs/authentication/production.
+// To customize the URL opener, or for more details on the URL format,
+// see URLOpener.
+//
+// See https://gocloud.dev/concepts/urls/ for background information.
 package cloudmysql // import "gocloud.dev/mysql/cloudmysql"
 
 import (
 	"context"
 	"database/sql"
 	"database/sql/driver"
 	"fmt"
+	"net/url"
+	"strings"
 	"sync"
 
 	"contrib.go.opencensus.io/integrations/ocsql"
-	"github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/certs"
 	"github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy"
 	"github.com/go-sql-driver/mysql"
 	"gocloud.dev/gcp"
 	"gocloud.dev/gcp/cloudsql"
-
-	// mysql enables use of the MySQL dialer for the Cloud SQL Proxy.
-	_ "github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/dialers/mysql"
+	cdkmysql "gocloud.dev/mysql"
 )
 
-// CertSourceSet is a Wire provider set that binds a Cloud SQL proxy
-// certificate source from an GCP-authenticated HTTP client.
-//
-// Deprecated: Use cloudsql.CertSourceSet.
-var CertSourceSet = cloudsql.CertSourceSet
+// Scheme is the URL scheme cloudmysql registers its URLOpener under on
+// mysql.DefaultMux.
+const Scheme = "cloudmysql"
```

Comment:
I wonder whether we should rethink the name of this package, maybe `gcpmysql`, `googlecloudmysql`, or `gcpcloudmysql`.

Reply:
@polinasok suggested that first in #2357; I'd rather do that in a separate PR (this already has a bunch going on). My vote is for `gcpmysql`.